### PR TITLE
[mypyc] Remangle redefined names produced by async with

### PR DESCRIPTION
**Pull request by ichard26**
Fixes mypyc/mypyc#1001.

### CODE REVIEW

File: mypyc/test-data/run-async.test [hunk 1]
```diff
@@ -143,3 +143,30 @@ async def foo() -> AsyncIterable[int]:
...
+class ConManB:
+    async def __aenter__(self) -> int:
+        return 2
+    async def __aexit__(self, *exc: object):
+        pass
+
+async def x() -> None:
+    value = 2
+    async with ConMan() as f:
+        value += f
```
Comment (JelleZijlstra) [inline mypyc/test-data/run-async.test:163]:
```suggestion
        value += f
    assert value == 3
```
Just check a little more precisely that the code is executing correctly.

File: mypyc/test-data/run-async.test [hunk 2]
```diff
@@ -143,3 +143,30 @@ async def foo() -> AsyncIterable[int]:
...
+    async with ConManB() as f:
+        value += f
+    assert value == 5
```
Comment (JelleZijlstra) [inline mypyc/test-data/run-async.test:166]:
```suggestion
    assert value == 5, value
```
So if this ever fails, we have a little more to go on.

**Comment by ichard26** [comment]
```
___________________________ mypy/test/testdaemon.py ___________________________
[gw0] win32 -- Python 3.8.10 D:\a\mypy\mypy\.tox\py38\Scripts\python.EXE
worker 'gw0' crashed while running 'mypy/test/testdaemon.py::DaemonSuite::daemon.test::testDaemonRun'
```
I love CI right now.

**Review by JukkaL — APPROVED** [review]
Thanks! Sorry for the slow review.

**Pull request merged**